sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -718,7 +717,8 @@ object ParquetFileFormat extends Logging {
   }
 
   /**
-   * ParquetFileFormat.prepareWrite calls this function to initialize `redirectParquetLogsViaSLF4J`.
+   * The ParquetFileFormat constructor calls this function to initialize
+   * `redirectParquetLogsViaSLF4J`.
    */
-  def redirectParquetLogs(): Unit = {}
+  private def redirectParquetLogs(): Unit = {}
```

Comment:
Maybe it would be clearer to call it once with an atomic boolean guard:

```
if (ParquetFileFormat.logsRedirected.compareAndSwap(false, true)) {
  ParquetFileFormat.redirectParquetLogsViaSLF4J()
}
```

Then you could test that the flag is set after constructing a ParquetFileFormat. That's obviously not a great test but probably good enough.